src/services/trace_service.py

```
@@ -305,6 +305,86 @@ def create_trace(
             logger.error(f"Message data that failed: {json.dumps(message_data, indent=2)[:500]}")
             return None
 
+    @staticmethod
+    def create_streaming_trace(
+        message_data: Dict[str, Any], instance_name: str, db_session: Session
+    ) -> Optional["StreamingTraceContext"]:
```

Ruff (0.12.2)

Undefined name `StreamingTraceContext`

(F821)